sdk/metricsadvisor/Azure.AI.MetricsAdvisor/tests/Models/DataFeedSourcesTests.cs

```diff
@@ -17,20 +17,36 @@ namespace Azure.AI.MetricsAdvisor.Tests
     {
-        private static object[] DataSourceTestCases =
+        private static object[] CreateDataSourceTestCases =
         {
-            new object[] { new AzureApplicationInsightsDataFeedSource("mock", "secret", "mock", "mock"), "\"apiKey\":\"secret\"" },
-            new object[] { new AzureBlobDataFeedSource("secret", "mock", "mock"), "\"connectionString\":\"secret\"" },
-            new object[] { new AzureCosmosDbDataFeedSource("secret", "mock", "mock", "mock"), "\"connectionString\":\"secret\"" },
-            new object[] { new AzureDataExplorerDataFeedSource("secret", "mock"), "\"connectionString\":\"secret\"" },
-            new object[] { new AzureDataLakeStorageGen2DataFeedSource("mock", "secret", "mock", "mock", "mock"), "\"accountKey\":\"secret\"" },
-            new object[] { new AzureTableDataFeedSource("secret", "mock", "mock"), "\"connectionString\":\"secret\"" },
-            new object[] { new InfluxDbDataFeedSource("secret", "mock", "mock", "mock", "mock"), "\"connectionString\":\"secret\"" },
-            new object[] { new InfluxDbDataFeedSource("mock", "mock", "mock", "secret", "mock"), "\"password\":\"secret\"" },
-            new object[] { new MongoDbDataFeedSource("secret", "mock", "mock"), "\"connectionString\":\"secret\"" },
-            new object[] { new MySqlDataFeedSource("secret", "mock"), "\"connectionString\":\"secret\"" },
-            new object[] { new PostgreSqlDataFeedSource("secret", "mock"), "\"connectionString\":\"secret\"" },
-            new object[] { new SqlServerDataFeedSource("secret", "mock"), "\"connectionString\":\"secret\"" }
+            new object[] { new AzureApplicationInsightsDataFeedSource("mock", "secret", "mock", "mock"), "apiKey" },
+            new object[] { new AzureBlobDataFeedSource("secret", "mock", "mock"), "connectionString" },
+            new object[] { new AzureCosmosDbDataFeedSource("secret", "mock", "mock", "mock"), "connectionString" },
+            new object[] { new AzureDataExplorerDataFeedSource("secret", "mock"), "connectionString" },
+            new object[] { new AzureDataLakeStorageGen2DataFeedSource("mock", "secret", "mock", "mock", "mock"), "accountKey" },
+            new object[] { new AzureTableDataFeedSource("secret", "mock", "mock"), "connectionString" },
+            new object[] { new InfluxDbDataFeedSource("secret", "mock", "mock", "mock", "mock"), "connectionString" },
+            new object[] { new InfluxDbDataFeedSource("mock", "mock", "mock", "secret", "mock"), "password" },
+            new object[] { new MongoDbDataFeedSource("secret", "mock", "mock"), "connectionString" },
+            new object[] { new MySqlDataFeedSource("secret", "mock"), "connectionString" },
+            new object[] { new PostgreSqlDataFeedSource("secret", "mock"), "connectionString" },
+            new object[] { new SqlServerDataFeedSource("secret", "mock"), "connectionString" }
+        };
+
+        private static object[] UpdateDataSourceTestCases =
```

Comment:
Wanted to find a way to unify `UpdateDataSourceTestCases` and `CreateDataSourceTestCases` in a single field, but it's not that simple. It seems NUnit instantiate the test case objects only once, so if we modify them in a test (Update test), the modified version may be passed to other tests and make them fail.

Reply:
Created https://github.com/Azure/azure-sdk-for-net/issues/21613 to investigate possible approaches.